integrationtests/self/handshake_test.go

```diff
@@ -140,6 +140,30 @@ var _ = Describe("Handshake tests", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 
+		It("has the right local and remote address on the ClientHelloInfo.Conn", func() {
```

Comment:
Now including an end-to-end test to make sure this API never accidentally breaks again.